docs/configuration/architecture_diff_func.md

```diff
@@ -0,0 +1,129 @@
+
+The diff function is explained below, and before using the diff functions, the two architecture objects are generated beforehand, as shown in the following script.
+
+```python
+from caret_analyze import Architecture
+left_arch = Architecture('yaml', 'old_architecture.yaml')
+right_arch = Architecture('yaml', 'new_architecture.yaml')
+
+```
+
+### diff_node_names()
+
+An example of the use of `diff_node_names()` is shown in `#sample_1`.
+
+<!-- Suppose arch1 has two nodes 'node1' and 'node2'. arch2 has three nodes 'node1', 'node3' and 'node4'.
+The type of the return value is `Tuple[Tuple[str, ...] , Tuple[str, ...]]` -->
+
+```python
+#sample_1
+
+#get node names that are only in old and new respectively.
+left_only_node_names, right_only_node_names = Architecture.diff_node_names(left_arch, right_arch)
+
+print(left_only_node_names)
+print(right_only_node_names)
+
+#result
+#('node2',)
+#('node3', 'node4')
+
+```
+
+In this setup, the `left_arch` has two nodes: `node1` and `node2`, and the `right_arch` has three nodes: `node1`, `node3` and `node4`. Therefore, the node name that `left_arch` and `right_arch` have in common is `node1`.
+Therefore, the node name that only `left_arch` has is `node2`, and the node names that only `right_arch` has are `node3` and `node4`, which are the return value and output at the end of the script respectively.
+
+### diff_topic_names()
+
+An example of the use of `diff_topic_names()` is shown in `#sample_2`.
+
+```python
+#sample_2
+
+#get topic names that are only in old and new respectively.
+left_only_topics, right_only_topics = Architecture.diff_topic_names(left_arch, right_arch)
+
+print(left_only_topics)
+print(right_only_topics)
+
+#result
+#('topic2',)
+#('topic3', 'topic4')
+
+```
+
+In this setup, the `left_arch` has two topic names: `topic1` and `topic2`, and the `right_arch` has three topic names: `topic1`, `topic3` and `topic4`. The topic name here is the name of the topic used for publish or subscribe.　Therefore, the topic name that `left_arch` and `right_arch` have in common is `topic1`.
+Therefore, the topic name that only `left_arch` has is `topic2`, and the topic names that only `right_arch` has are `topic3` and `topic4`, which are the return value and output at the end of the script respectively.
+
+## How to compare node in architecture
+
+<!-- This section describes the functions `diff_node_pubs()` and `diff_node_subs()`, which compare the contents of two nodes in architecture. -->
+
+`diff_node_pubs()` and `diff_node_subs()` are functions that take nodes in an architecture and compare them. The nodes given can be two nodes in the same architecture or in two different architectures. These functions compare the two given nodes and find the pub/sub topic name that exists only in each of them. Thus, what one node has but the other does not (pub/sub topic name) can be identified and compared.
```

Comment:
```suggestion
Nodes consist of the node name, publish topic, and subscription topic.
The `diff_node_pubs()` and `diff_node_subs()` function find the publish and subscription topics that exist only in one of the two nodes. 

```

Reply:
Don't need to apply this modification.
Please apply [here](https://github.com/tier4/CARET_doc/pull/168#pullrequestreview-1347730340).